apps/aeoracle/src/aeo_state_tree.erl

```diff
@@ -0,0 +1,83 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2017, Aeternity Anstalt
+%%% @doc
+%%% ADT for keeping the state of oracles
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aeo_state_tree).
+
+%% API
+-export([ get_interaction/2
+        , get_oracle/2
+        , find_interaction/2
+        , find_oracle/2
+        , new/0
+        , put_interaction/2
+        , put_oracle/2
+        ]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-type tree() :: gb_merkle_trees:tree().
+-type interaction() :: aeo_interaction:interaction().
+-type oracle() :: aeo_oracles:oracle().
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec new() -> tree().
+new() ->
+    {ok, Tree} = aec_trees:new_merkle_tree(),
```

Comment:
I see this PR contains portions of oracles. Are these intended to be included in the same PR?
I see the oracles tree conflicts with PR #505.